internal/controllers/reference_addon_controller.go

```diff
@@ -2,35 +2,63 @@ package controllers
 
 import (
 	"context"
-	"strings"
 
 	"github.com/go-logr/logr"
+	"k8s.io/apimachinery/pkg/api/errors"
 	"k8s.io/apimachinery/pkg/runtime"
 	ctrl "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 
 	refapisv1alpha1 "github.com/openshift/reference-addon/apis/reference/v1alpha1"
+	"github.com/openshift/reference-addon/internal/addonsdk"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
 type ReferenceAddonReconciler struct {
 	client.Client
-	Log    logr.Logger
-	Scheme *runtime.Scheme
+	StatusReporter *addonsdk.StatusReporter
+	Log            logr.Logger
+	Scheme         *runtime.Scheme
 }
 
 func (r *ReferenceAddonReconciler) Reconcile(
 	ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
-	log := r.Log.WithValues("addon", req.NamespacedName.String())
+	log := r.Log.WithValues("referenceaddon", req.NamespacedName.String())
+
+	successfulCondition := metav1.Condition{
+		Type:    addonsdk.AddonHealthyConditionType,
+		Status:  metav1.ConditionTrue,
+		Reason:  "AllComponentsUp",
+		Message: "Everything under reference-addon is working perfectly fine",
```

Comment:
Should refer to this just being a result of the property in the `ReferenceAddon` API.